android/app/src/main/java/org/uwaterloo/subletr/pages/home/list/components/RoomsFilter.kt

```diff
@@ -0,0 +1,232 @@
+package org.uwaterloo.subletr.pages.home.list.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material3.SwitchDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.components.switch.PrimarySwitch
+import org.uwaterloo.subletr.pages.home.list.HomeListUiState
+import org.uwaterloo.subletr.theme.filterRegularFont
+import org.uwaterloo.subletr.theme.secondaryButtonBackgroundColor
+import org.uwaterloo.subletr.theme.secondaryTextColor
+import org.uwaterloo.subletr.theme.subletrPink
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+
+@Composable
+fun RoomFilter(
+	currentRoomRange: HomeListUiState.RoomRange,
+	updateRoomFilter: (HomeListUiState.RoomRange) -> Unit,
+	closeAction: () -> Unit,
+) {
+	var bedroomForSublet by remember {
+		mutableStateOf(
+			currentRoomRange.bedroomForSublet
+		)
+	}
+	var bedroomInProperty by remember {
+		mutableStateOf(
+			currentRoomRange.bedroomInProperty
+		)
+	}
+	var bathroom by remember {
+		mutableStateOf(
+			currentRoomRange.bathroom
+		)
+	}
+	var ensuitebathroom by remember {
+		mutableStateOf(
+			currentRoomRange.ensuiteBathroom
+		)
+	}
+	BasicFilterLayout(
+		modifier = Modifier.height(500.dp),
```

Comment:
Do you need this or can we make it wrap content height?

Reply:
No...